### ISSUE TITLE
Spatial type support ?

### ISSUE THREAD

**Issue by carlwoodhouse** [sebastienros/yessql#198]
Would it be possible to get support for spatial types ? (eg sql geography) and queries ?

Not sure how you'd go about doing it for each provider, i think EFcore abstracts it using https://github.com/NetTopologySuite/NetTopologySuite - so maybe something to look at ?

**Comment by sebastienros** [comment]
https://www.npgsql.org/efcore/mapping/nts.html
https://www.markopapic.com/finding-nearby-users-using-ef-core-spatial-data/
https://docs.microsoft.com/en-us/ef/core/modeling/spatial

That should work the same way. I think this should be easy to implement, just that the spatial types need to be mapped in the dialects. Then each custom method in the predicates can be handled the same way we already handle `String.Contains` or `IsIn` ...

**Comment by carlwoodhouse** [comment]
@sebastienros Got to a point where i could really use this; so started looking at it ...

Ran into an issue that the dialects override `GetTypeName` which expects a `DbType`; the spatial types arent listed in these those :( so not sure of the best way to achieve the mapping, any thoughts ?

**Comment by sebastienros** [comment]
Just looked. I use DbType in the dialects as a way to normalize which types are supported in the dbs. And in `SchemaUtils` there is a helper to convert a clr type to DbType. I understand you are limited by the DbType enumeration. In that case I suggest that you create a new enum, and replace DbType usage with this new one. Then you can add the elementary types that can be mapped in the databases. Then you can also change the SchemaUtils to understand the types from each external package. Maybe by dynamic registration such that each db package handles that (postgre, sqlserver, ...).

**Comment by carlwoodhouse** [comment]
Yeah i was thinking a new enum would be the way to go; seemed a little drastic but i can't see an alternative!
